orange-tree.rb

```diff
@@ -0,0 +1,132 @@
+# this one is another group project! three adies to a group. :)
+# https://github.com/Ada-Developers-Academy/daily-curriculum/blob/master/topic_resources/orange_tree.md
+# -------------------------------------
+# actually, according to the new dinner requirements, I think we can move on to dessert!
+
+class OrangeTree
+
+  attr_reader :age, :alive, :grove, :height, :orange_count
+
+  def initialize
+    @age = 0
+    @alive = true
+    @height = 0
+    @orange_count = 0
+  end
+
+  def grow_oranges
+    if @age >= 3 # oranges don't grow on young trees!
+      if @grove.soil_quality == "unhealthy"
```

Comment:
While trying to run this program in IRB I found that it errored here because it does not know about @grove at this point in the program. This is probably part of the bugginess you were referring to.

There is this concept in programming that you will hear a lot about over time called the Law of Demeter: http://en.wikipedia.org/wiki/Law_of_Demeter

Essentially you think to yourself what it actually makes sense for your object to know about and limiting its interface to that. So think a bit anthropomorphically of Trees and Groves for a second with me. Does it make sense to ask a tree about the soil quality of the entire grove it resides in? Not really. It only knows its own context. Does it make sense to ask a tree about the soil quality immediately around its roots? Assuming you've made the anthropomorphic tree leap with me, then yes it does.

So consider trying to have soil quality be something that is applied to trees from the grove class instead.